### ISSUE TITLE
[Bug]: Upgrading the timescale extension doesn't work after downgrading it.

### ISSUE THREAD

**Issue by varunsaral** [timescale/timescaledb#6496]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Other

### What happened?

After downgrading to version 2.12.2 from 2.13.0 i was trying to upgrade it again to 2.13.0 and was facing `catalog version mismatch expected '2.12.2' seen '2.13.0'`. After deleting the entry from _timescaledb_catalog.metadata it started working

### TimescaleDB version affected

2.13.9

### PostgreSQL version used

14.0

### What operating system did you use?

ubuntu

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
Downgrade from version 2.13.0 to 2.12.0 and upgrade it again
```

**Comment by jnidzwetzki** [comment]
Hello @varunsaral,

Thank you for reaching out and for the detailed steps to reproduce the problem. I could reproduce the problem locally with TimescaleDB 2.12.2 and TimescaleDB 2.13.0.

```sql
create extension timescaledb version '2.13.0';
[...]
alter extension timescaledb update to '2.12.2';
[...]
testupgrade=# alter extension timescaledb update to '2.13.0';
ERROR:  catalog version mismatch, expected "2.12.2" seen "2.13.0"
CONTEXT:  PL/pgSQL function inline_code_block line 7 at RAISE
```

In contrast, installing TimescaleDB 2.12.2 and upgrading it to 2.13.0 works without any problem.

**Comment by jnidzwetzki** [comment]
Hello @varunsaral,

The issue is fixed beginning with TimescaleDB 2.13.1. For up- and downgrading version 2.13.0, the corresponding entry in `_timescaledb_catalog.metadata` must be deleted as a workaround before.

```sql
# delete from _timescaledb_catalog.metadata where key = 'timescaledb_version' and value = '2.13.0';
# [Reconnect]
# alter extension timescaledb update to '2.13.0';

# \dx
                                                List of installed extensions
    Name     | Version |   Schema   |                                      Description                                      
-------------+---------+------------+---------------------------------------------------------------------------------------
 plpgsql     | 1.0     | pg_catalog | PL/pgSQL procedural language
 timescaledb | 2.13.0  | public     | Enables scalable inserts and complex queries for time-series data (Community Edition)
(2 rows)
```

**Comment by jflambert** [comment]
@jnidzwetzki didn't try 2.13.1 but heads up that this issue #6496 is not in the tag changelog...

https://github.com/timescale/timescaledb/releases/tag/2.13.1
